pkg/controller/acmeorders/sync.go

```diff
@@ -167,16 +176,12 @@ func (c *controller) Sync(ctx context.Context, o *cmacme.Order) (err error) {
 	}
 
 	switch {
-	case o.Status.State == cmacme.Ready:
```

Comment:
Why remove this check? There could be cases where we don't have `Ready` persisted on the CRD resource, but the actual Order in the ACME server *is* Ready, right? Or do we want to enforce multiple reconciles in this case? (e.g. we don't finalize until we have `Ready` persisted onto our own representation?)

Reply:
Unless I'm missing something this shouldn't make any difference as I just moved that check up to avoid the extra call to the ACME server.
Before this was
```
	switch {
	case o.Status.State == cmacme.Ready:
		log.V(logf.DebugLevel).Info("Finalizing Order as order state is 'Ready'")
		return c.finalizeOrder(ctx, cl, o, genericIssuer)
....
```
and now it is 
```
	if o.Status.State == cmacme.Ready {
		log.V(logf.DebugLevel).Info("Finalizing Order as order state is 'Ready'")
		return c.finalizeOrder(ctx, cl, o, genericIssuer)
	}
// do the call to get ACME order
switch {
// all other cases
```

Reply:
I'll check at which point we set the order status on the CRD

Reply:
> Or do we want to enforce multiple reconciles in this case?

I think we do multiple reconciles at this point. I assume that the 'ready' state would likely be set in [this case](https://github.com/jetstack/cert-manager/blob/master/pkg/controller/acmeorders/sync.go#L217-L229).
I guess that we do want to have two reconciles there else we'd skip to show the 'ready' step on the 'Order' altogether and attempt to move straight to 'valid' by immediately trying to finalize the order once all challenges are completed.

Reply:
> (e.g. we don't finalize until we have Ready persisted onto our own representation?)

I think successful finalize should end up in 'Valid' state, 'Ready' is a intermediate state